src/impl.jl

```diff
@@ -249,16 +264,34 @@ function auto_hash_equals_impl(__source__, struct_decl, fields, cache::Bool, has
             x._cached_hash
         end)))
     else
-        hash_typearg = typearg && !isnothing(where_list)
-        hash_init = hash_typearg ? :($hashfn($full_type_name, h)) : :($hashfn($(QuoteNode(type_name)), h))
+        hash_init =
+            if isnothing(typeseed)
+                if typearg
+                    :($hashfn($full_type_name, h))
+                else
+                    :($hashfn($(QuoteNode(type_name)), h))
+                end
+            else
+                if typearg
+                    :(h + UInt($typeseed($full_type_name)))
+                else
+                    :(h + UInt($typeseed))
```

Comment:
is it worth helpering this chunk as it's also used above?

Reply:
They are subtly different.  This latter version has `h`, the former does not.  Every code line is different.